tsl/src/compression/compression.c

```diff
@@ -1512,3 +1516,155 @@ update_compressed_chunk_relstats(Oid uncompressed_relid, Oid compressed_relid)
 		CommandCounterIncrement();
 	}
 }
+
+typedef struct CompressSingleRowState
+{
+	Relation in_rel;
+	Relation out_rel;
+	RowCompressor row_compressor;
+	TupleTableSlot *out_slot;
+} CompressSingleRowState;
+
+static TupleTableSlot *compress_singlerow(CompressSingleRowState *cr, TupleTableSlot *in_slot);
+
+CompressSingleRowState *
+compress_row_init(int srcht_id, Relation in_rel, Relation out_rel)
+{
+	ListCell *lc;
+	List *htcols_list = NIL;
+	int i = 0, cclen;
+	const ColumnCompressionInfo **ccinfo;
+	TupleDesc in_desc = RelationGetDescr(in_rel);
+	TupleDesc out_desc = RelationGetDescr(out_rel);
+	int16 *in_column_offsets;
+	int n_keys;
+	const ColumnCompressionInfo **keys;
+
+	CompressSingleRowState *cr = palloc(sizeof(CompressSingleRowState));
```

Comment:
Probably safer to use `palloc0` since if fields are added, they will get random values and we might not notice it.

Reply:
Unless you make a mistake when adding a field, which happens.